interview-questions.md

```diff
@@ -10,42 +10,69 @@ Then, research the question to expand on your answer. Even if you feel you have
+
 Researched answer:
 
+Ruby on rails is a server side web application framework. It is based on a model view controller architecture pattern. MVC separates data structures from the User Interface design and offers different views of data. Model is the database layer. the view is the part the user sees. the controller is the switch board that collects data from model, then send it to construct the view. 
+
+
 4. What is a relational database? Are there other kinds of databases?
 
 Your answer:
 
+Relational database is the process of information being entered in Rails table and columns.
+
 Researched answer:
 
+A relational database is a type of database that store access to data points. Relational database includes tables containing rows and columns. Another kind of database is SQLite. SQlite is used when creating a new project. 
+
+
 5. What are primary keys? Why are they important?
 
 Your answer:
 
+Primary keys in Ruby are similar to keywords in javscript. They are important because they make the life of the developers smoother.
+
 Researched answer:
 
+A primary key is a key in a relational database that is unique for each record. Primary are also called a primary keyword. It is an unique identifier, like telephone number, driver license number or social security number etc...
+Primary keys are important because they identify important information. 
+
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-1. RESTful routes:
+1. RESTful routes: A RESTful route is a route that provides mapping from HTTP verbs (get, post, put, delete, patch) to controller CRUD actions (create, read, update, delete). Instead of relying solely on the URL to indicate what site to visit, a RESTful route depends on the HTTP verb and the URL.
 
-2. JSON:
+2. JSON: JavaScript Object Notation, more commonly known by the acronym JSON, is an open data interchange format that is both human and machine-readable. Despite the name JavaScript Object Notation, JSON is independent of any programming language and is a common API output in a wide variety of applications.
 
-3. ERB:
+3. ERB: ERB (or Ruby code generated by ERB ) returns a string in the same character encoding as the input string. When the input string has a magic comment, however, it returns a string in the encoding specified by the magic comment.
 
-4. Params:
+4. Params: params is an alias for the parameters method. params comes from ActionController::Base, which is accessed by your application via ApplicationController. Specifically, params refers to the parameters being passed to the controller via a GET or POST request.
 
-5. API:
+5. API: API is the acronym for Application Programming Interface, which is a software intermediary that allows two applications to talk to each other. Each time you use an app like Facebook, send an instant message, or check the weather on your phone, you're using an API.
```

Comment:
excellent work on the interview questions.  Love your examples.  Keep up the great work!